hermitian-XRCC/SH_diagrams.py

```diff
@@ -89,35 +133,31 @@ class body_2(object):
         result00 = body_2._H1_one_body00(densities, integrals, subsystem, charges, permutation=(0,1))
-        result01 = body_2._H1(densities, integrals, subsystem, charges, permutation=(0,1))
-        result10 = body_2._H1(densities, integrals, subsystem, charges, permutation=(1,0))
+        result01 = body_2._H1(           densities, integrals, subsystem, charges, permutation=(0,1))
+        result10 = body_2._H1(           densities, integrals, subsystem, charges, permutation=(1,0))
         result11 = body_2._H1_one_body00(densities, integrals, subsystem, charges, permutation=(1,0))
         return [result00, result01, result10, result11]
     @staticmethod
     def _H1(densities, integrals, subsystem, charges, permutation):
-        rho1, rho2, n_i2, p, (chg_i1,chg_j1), (chg_i2,chg_j2) = _parameters2(densities, subsystem, charges, permutation)
-        h01 = _ints2(subsystem, permutation, integrals, ["h01"])
-        prefactor = (-1)**(n_i2 + p)
-        if chg_i1==chg_j1-1 and chg_i2==chg_j2+1:
-            c1 = rho1['c'][chg_i1,chg_j1]
-            a2 = rho2['a'][chg_i2,chg_j2]
-            def diagram(i1,i2,j1,j2):
-                #return prefactor * np.einsum("pq,p,q->", h01, c1[i1][j1], a2[i2][j2])
-                partial = tendot(h01, tl.tensor(c1[i1][j1]), axes=([0], [0]))
-                return prefactor * tendot(partial, tl.tensor(a2[i2][j2]), axes=([0], [0]))
+        X = _parameters(densities, integrals, subsystem, charges, permutation)
+        if X.Dchg_0==-1 and X.Dchg_1==+1:
+            prefactor = (-1)**(X.n_i1 + X.P)
+            def diagram(i0,i1,j0,j1):
+                #return prefactor * numpy.einsum("pq,p,q->", h01, X.c_0[i0][j0], X.a_1[i1][j1])
+                partial =          tendot(X.h_01,  X.c_0[i0][j0], axes=([0], [0]))
+                return prefactor * tendot(partial, X.a_1[i1][j1], axes=([0], [0]))
             return diagram, permutation
         else:
             return None, None
     @staticmethod
     def _H1_one_body00(densities, integrals, subsystem, charges, permutation):
-        rho1, rho2, n_i2, p, (chg_i1,chg_j1), (chg_i2,chg_j2) = _parameters2(densities, subsystem, charges, permutation)
-        if chg_i1==chg_j1 and chg_i2==chg_j2:
-            h00 = _ints2(subsystem, permutation, integrals, ["h00"])
-            ca1 = rho1["ca"][chg_i1,chg_j1]
-            def diagram(i1,i2,j1,j2):
-                if i2==j2:
-                    #return np.einsum("pq,pq->", h00, ca1[i1][j1])
-                    return tendot(h00, tl.tensor(ca1[i1][j1]), axes=([0, 1], [0, 1]))
+        X = _parameters(densities, integrals, subsystem, charges, permutation)
+        if X.Dchg_0==0 and X.Dchg_1==0:
+            prefactor = 1
+            def diagram(i0,i1,j0,j1):
+                if i1==j1:
+                    #return numpy.einsum("pq,pq->", h00, X.ca_0[i0][j0])
+                    return prefactor * tendot(X.h_00, X.ca_0[i0][j0], axes=([0, 1], [0, 1]))
```

Comment:
Did you see this comment in _parameters?:
# All the if switches here can go away when the top-level drivers are unified
I only did it this mis-matched way for now so that everything could be tested using the same code stack.  After I merge the top levels, I will get rid of this.  I'll ask if it is not obvious what the best way is.